video/out/opengl/context_drm_egl.c

```diff
@@ -388,27 +388,17 @@ static void wait_fence(struct ra_ctx *ctx)
     }
 }
 
-static bool drm_egl_start_frame(struct ra_swapchain *sw, struct ra_fbo *out_fbo)
-{
-    struct ra_ctx *ctx = sw->ctx;
-    struct priv *p = ctx->priv;
-    struct vo_drm_state *drm = ctx->vo->drm;
-
-    if (!drm->atomic_context->request) {
-        drm->atomic_context->request = drmModeAtomicAlloc();
-        p->drm_params.atomic_request_ptr = &drm->atomic_context->request;
-    }
-
-    return ra_gl_ctx_start_frame(sw, out_fbo);
-}
-
 static bool drm_egl_submit_frame(struct ra_swapchain *sw, const struct vo_frame *frame)
 {
     struct ra_ctx *ctx = sw->ctx;
     struct vo_drm_state *drm = ctx->vo->drm;
 
     drm->still = frame->still;
 
+    // With gpu-next, only the drm state needs to be updated.
+    if (strcmp(ctx->vo->driver->name, "gpu-next") == 0)
```

Comment:
Can we instead strcmp add in vo.c something like
```
bool vo_is_submit_frame_needed(struct vo *vo)
{
    return vo->driver == &video_out_gpu_next;
}
```

Reply:
Sure